ethers-solc/src/cache.rs

```diff
@@ -411,8 +448,8 @@ pub struct CacheEntry {
     pub content_hash: String,
     /// identifier name see [`crate::utils::source_name()`]
     pub source_name: PathBuf,
-    /// what config was set when compiling this file
-    pub solc_config: SolcConfig,
+    /// what compilation unit this file was a part of
+    pub compilation_unit: CompilationUnitId,
```

Comment:
@gakonst One thing I realized chasing down #1995 is that each cache entry should have multiple compilation units instead of just one so maybe the compilation unit should be stored at the artifact level. This would allow the cache to handle having the same version being compiled with different compiler settings e.g. someone doesn't want to use via-IR while testing. I explore this in a separate branch but not sure it's a great solution (https://github.com/0xalpharush/ethers-rs/commit/c378b495f712aae2cc077d6d3e50ab4def1e67bd).

Reply:
this is a reasonable consideration and would give us a bit more flexibility I think.

can't think of any downsides to moving this to the artifact level

Reply:
SGTM.